.github/workflows/test-chart.yml

```diff
@@ -0,0 +1,102 @@
+    name: Run on Ubuntu
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone the code
+        uses: actions/checkout@v4
+
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+
+      - name: Install the latest version of kind
+        run: |
+          curl -Lo ./kind https://kind.sigs.k8s.io/dl/latest/kind-linux-amd64
+          chmod +x ./kind
+          sudo mv ./kind /usr/local/bin/kind
+
+      - name: Verify kind installation
+        run: kind version
+
+      - name: Create kind cluster
+        run: kind create cluster
+
+      - name: Prepare integration-service
+        run: |
+          go mod tidy
+          export IMG=integration-service:v$DUMMY_VERSION
+          yq -i '.controllerManager.container.image = env(IMG)' dist/chart/values.yaml
+          make img-build IMG=$IMG
+          kind load docker-image $IMG
+
+      - name: Verify Helm installation
+        run: helm version
+
+      - name: Lint Helm Chart
+        run: |
+          helm lint ./dist/chart
+
+      - name: Install cert-manager via Helm
+        run: |
+          helm repo add jetstack https://charts.jetstack.io
+          helm repo update
+          helm install cert-manager jetstack/cert-manager --namespace cert-manager --create-namespace --set installCRDs=true
+
+      - name: Wait for cert-manager to be ready
+        run: |
+          kubectl wait --namespace cert-manager --for=condition=available --timeout=300s deployment/cert-manager
+          kubectl wait --namespace cert-manager --for=condition=available --timeout=300s deployment/cert-manager-cainjector
+          kubectl wait --namespace cert-manager --for=condition=available --timeout=300s deployment/cert-manager-webhook
+
+      - name: Install Prometheus Operator CRDs
+        run: |
+          helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
+          helm repo update
+          helm install prometheus-crds prometheus-community/prometheus-operator-crds
+
+      - name: Install Prometheus via Helm
+        run: |
+          helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
+          helm repo update
+          helm install prometheus prometheus-community/prometheus --namespace monitoring --create-namespace
+
+      - name: Wait for Prometheus to be ready
+        run: |
+          kubectl wait --namespace monitoring --for=condition=available --timeout=300s deployment/prometheus-server
+
+      - name: Install Helm chart for project
+        run: |
+          helm upgrade --install my-release ./dist/chart --set chartCreateNamespace=true
+
+      - name: Check Helm release status
+        run: |
+          helm status my-release
+
+      - name: Check Presence of ServiceMonitor
+        run: |
+          kubectl wait --namespace integration-service --for=jsonpath='{.kind}'=ServiceMonitor servicemonitor/integration-service-controller-manager-metrics-monitor
```

Comment:
No timeout is needed here?

Reply:
It's using the default 30s. I'll add that for consistency.